poly-synth/src/poly-synth-voice.js

```diff
@@ -58,12 +58,35 @@ class PolySynthVoice {
   start() {
     // Ramp to full amplitude in attack (s) and to sustain in decay (s).
     const t = this.context_.currentTime;
-    const timeToFullAmplitude = t + this.parameters_.attack;
-    const timeToSustain = timeToFullAmplitude + this.parameters_.decay;
+    const timeToFullAmplitude = t + this.parameters_.gainAttack;
+
+    // |timeToGainSustain| must be later than |timeToFullAmplitude|.
+    const timeToGainSustain =
+        timeToFullAmplitude + this.parameters_.gainDecay + 0.001;
```

Comment:
The reason I added a small constant to` timeToFullAmplitude + this.parameters_.gainDecay` is that timeToGainSustain cannot be the same as timeToFullAmplitude since the two linearRampToValueAtTime commands will conflict. Thinking some more about it, the best solution seems to me to be to set the minimum decay time to a small positive value.